### PR TITLE
S3 tests

### PR DESCRIPTION
**Pull request by Jamstah**
The s3 tests were failing for numerous reasons, so this PR fixes them.

Long term it would be better to mock the s3 SDK so that everyone could run them, and to run AWS connected tests in the CICD pipeline.

### CODE REVIEW

**Comment by Jamstah** [comment]
Rebased. For whatever reason, it seems I'm now considered a first time contributor again, so can I please have an OK to test?

File: registry/storage/driver/s3-aws/s3_test.go [hunk 1]
```diff
@@ -23,13 +21,8 @@ import (
...
 
-// Hook up gocheck into the "go test" runner.
-func Test(t *testing.T) { check.TestingT(t) }
-
-var (
-	s3DriverConstructor func(rootDirectory, storageClass string) (*Driver, error)
-	skipS3              func() string
-)
+var s3DriverConstructor func(rootDirectory, storageClass string) (*Driver, error)
+var skipS3 func() string
```
Comment (milosgajdos) [inline registry/storage/driver/s3-aws/s3_test.go:25]: Any reason why these are no longer grouped?

Comment (milosgajdos) [reply registry/storage/driver/s3-aws/s3_test.go:25]: I dont have a preference per se, just noticed it in the diff and was curious about why

**Comment by milosgajdos** [comment]
> Long term it would be better to mock the s3 SDK so that everyone could run them, and to run AWS connected tests in the CICD pipeline.

Mocking sounds good and I'm all for it but it merely covers unit tests - we need those indeed. I'm more keen on using some S3-compatible API we could stand up during the tests. Something like `minio` or some such - maybe `localstack` (alas this is slow).

**Review by milosgajdos — APPROVED** [review]

**Comment by milosgajdos** [comment]
@Jamstah could you rebase, please

**Comment by DavidSpek** [comment]
I've looked at the code quickly but want to do so more in depth tomorrow before I add my review. Just wanted to let you know I am busy with it.

**Review by DavidSpek — APPROVED** [review]
LGTM

**Pull request merged**